core-java-modules/core-java-collections-list-7/src/test/java/com/baeldung/sortlistofpair/SortListOfPairUnitTest.java

```diff
@@ -0,0 +1,78 @@
+package com.baeldung.sortlistofpair;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.Comparator;
+import java.util.List;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.junit.jupiter.api.Test;
+
+import com.google.common.collect.Lists;
+
+public class SortListOfPairUnitTest {
+
+    private List<Pair<String, Integer>> getUnsortedInput() {
+        return Lists.newArrayList(
+            //@formatter:off
+            Pair.of("a b c d e",5),
+            Pair.of("a b c",3),
+            Pair.of("a b c d",4),
+            Pair.of("a b",2),
+            Pair.of("a",1)
+            //@formatter:on
+        );
+    }
+
+    private static final List<Pair<String, Integer>> EXPECTED = List.of(
+        //@formatter:off
+        Pair.of("a",1),
+        Pair.of("a b",2),
+        Pair.of("a b c",3),
+        Pair.of("a b c d",4),
+        Pair.of("a b c d e",5)
+        //@formatter:on
+    );
+
+    @Test
+    void whenSortingUsingAnonymousComparator_thenCorrect() {
+        List<Pair<String, Integer>> myList = getUnsortedInput();
+        myList.sort(new Comparator<Pair<String, Integer>>() {
+            @Override
+            public int compare(Pair<String, Integer> o1, Pair<String, Integer> o2) {
+                return o1.getRight()
+                    .compareTo(o2.getRight());
+            }
+        });
+
+        assertEquals(EXPECTED, myList);
+    }
+
+    @Test
+    void whenSortingUsingLamda_thenCorrect() {
+        List<Pair<String, Integer>> myList = getUnsortedInput();
+        myList.sort((p1, p2) -> p1.getRight()
+            .compareTo(p2.getRight()));
+
+        assertEquals(EXPECTED, myList);
+    }
+
+    @Test
+    void whenSortingUsingComparatorComparing_thenCorrect() {
+        List<Pair<String, Integer>> myList = getUnsortedInput();
+        myList.sort(Comparator.comparing(Pair::getRight));
+
+        assertEquals(EXPECTED, myList);
+    }
+
+    @Test
+    void whenSortingUsingStream_thenCorrect() {
+        List<Pair<String, Integer>> unsorted = List.copyOf(getUnsortedInput());
```

Comment:
Why are we doing List.copyOf here? This seems redundant since we use .stream()

Reply:
You are right. Using List.copyOf() to explicitly build an immutable list.
getunsortedInput() returns a mutable list.